docs/architecture/types/README.md

```diff
@@ -200,6 +257,41 @@ Output<T>>`). In this manner, input properties may accept either plain values
 defined in the program outright, or values which arise from the outputs of other
 resources.
 
+(property-paths)=
+### Property paths
+
+A *property path* is [JSONPath](https://en.wikipedia.org/wiki/JSONPath)-like
+expression that describes a path to one or more properties within a set of
+values, such as those that may comprise a resource's [input](inputs) or
+[output](outputs) properties. Property paths are used in many contexts, such as
+specifying a set of properties to
+[`ignoreChanges`](https://www.pulumi.com/docs/iac/concepts/options/ignorechanges/)
+for, or when identifying the set of properties responsible for a particular
+[](pulumirpc.DiffResponse). Example property paths include:
+
+```
+root
+root.nested
+root["nested"]
+root.double.nest
+root["double"].nest
+root["double"]["nest"]
+root.array[0]
+root.array[100]
+root.array[0].nested
+root.array[0][1].nested
+root.nested.array[0].double[1]
+root["key with \"escaped\" quotes"]
+root["key with a ."]
+["root key with \"escaped\" quotes"].nested
+["root key with a ."][100]
+root.array[*].field
```

Comment:
These are now property path patterns instead of property paths right? Could link to the use of these in ignoreChanges.

Reply:
Yep I've added this section precisely so that `ignoreChanges` docs can link to it :smile: